ni_measurementlink_service/_internal/parameter/metadata.py

```diff
@@ -53,17 +58,76 @@ def validate_default_value_type(parameter_metadata: ParameterMetadata) -> None:
-            if not isinstance(element, expected_element_type):
+        if has_enum_values_annotation:
+            for element in default_value:
+                if not _is_default_enum_match_annotations(element, enum_values_annotation):
+                    raise TypeError(
+                        f"The default value, `{element}`, for '{display_name}' is not valid. Expected values: `{enum_values_annotation}`."
+                    )
+        else:
+            for element in default_value:
+                if not isinstance(element, expected_element_type):
+                    raise TypeError(
+                        f"Unexpected element of type {type(element)} in the default value for '{display_name}'. Expected element type: {expected_element_type}."
+                    )
+    else:
+        if has_enum_values_annotation:
+            if not _is_default_enum_match_annotations(default_value, enum_values_annotation):
+                raise TypeError(
+                    f"The default value, `{default_value}`, for '{display_name}' is not valid. Expected values: `{enum_values_annotation}`."
+                )
+        else:
+            if not isinstance(default_value, expected_type):
                 raise TypeError(
-                    f"Unexpected element of type {type(element)} in the default value for '{display_name}'. Expected element type: {expected_element_type}."
+                    f"Unexpected type {type(default_value)} in the default value for '{display_name}'. Expected type: {expected_type}."
                 )
     return None
+
+
+def try_get_enum_values_annotation(parameter_metadata: ParameterMetadata) -> Tuple[bool, str]:
+    """Gets the value for the "ni/enum.values" annotation if it exists.
+
+    Args
+    ----
+        parameter_metadata (ParameterMetadata): Parameter metadata
+
+    Returns
+    -------
+        Tuple[bool, str]: A Tuple that contains both the value of
+        the "ni/enum.values" annotation and whether we were successful
+        in obtaining it.
+
+    """
+    if (
+        "ni/type_specialization" in parameter_metadata.annotations
+        and parameter_metadata.annotations["ni/type_specialization"]
+        == TypeSpecialization.Enum.value
+        and "ni/enum.values" in parameter_metadata.annotations
+    ):
+        enum_values = parameter_metadata.annotations["ni/enum.values"]
+        if enum_values != "" and enum_values is not None:
+            return True, enum_values
+    return False, ""
+
+
+def _is_default_enum_match_annotations(default_value: Any, enum_values_annotation: str):
+    if not isinstance(default_value, Enum):
+        return False
+    user_enum = json.loads(enum_values_annotation.replace("'", '"'))
+    user_enum_class = Enum("UserDefinedEnum", user_enum)
+    if (
+        any(member.value == default_value.value for member in user_enum_class)
+        and user_enum_class(default_value.value).name == default_value.name
+    ):
+        return True
+    return False
```

Comment:
`json.loads()` returns a dict. Why construct a temporary enum class rather than validate against the dict? Does it handle some Python-specific enum naming special case? Or is this to look up keys by value?

Reply:
If the point is to verify that both the name and value are in the enum annotation and not some `Enum` class specific behavior, this should work:
```
user_enum = json.loads(enum_values_annotation.replace("'", '"'))
return default_value.name in user_enum and user_enum[default_value.name] == default_value.value
```

Reply:
Removed temporary enum class as it is unnecessary. I had not realized that `json.loads` returned an easy-to-use dict and thought it necessary to create an enum class to easily validate that the default value matched the enum information from the annotations.

Reply:
Didn't see your latest comment. I updated to take your simplified logic as suggested.